pkg/controllers/nodeclass/ami.go

```diff
@@ -47,6 +47,10 @@ func NewAMIReconciler(provider amifamily.Provider) *AMI {
 func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (reconcile.Result, error) {
 	amis, err := a.amiProvider.List(ctx, nodeClass)
 	if err != nil {
+		if amifamily.IsAl2DeprecationError(err) {
+			nodeClass.StatusConditions().SetFalse(v1.ConditionTypeAMIsReady, "AL2Deprecation", err.Error())
```

Comment:
nit: I'd use a more generic term for the reason, maybe something along the lines of `UnsupportedAlias`. The additional AL2 information will be propagated through the message.